packages/kbn-generate-csv/src/lib/search_cursor.ts

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+import type { IScopedClusterClient, Logger } from '@kbn/core/server';
+import type {
+  IEsSearchResponse,
+  IKibanaSearchResponse,
+  ISearchClient,
+  ISearchSource,
+  SearchRequest,
+  SearchSourceFields,
+} from '@kbn/data-plugin/common';
+import { ES_SEARCH_STRATEGY } from '@kbn/data-plugin/common';
+import { lastValueFrom } from 'rxjs';
+import type { CsvPagingStrategy } from '../../types';
+import type { CsvExportSettings } from './get_export_settings';
+import { i18nTexts } from './i18n_texts';
+
+interface Clients {
+  data: ISearchClient;
+  es: IScopedClusterClient;
+}
+
+export type SearchCursorSettings = Pick<
+  CsvExportSettings,
+  'scroll' | 'includeFrozen' | 'maxConcurrentShardRequests'
+>;
+
+export class SearchCursor {
```

Comment:
just a suggestion: Have you considered splitting this into two separate objects? e.g. ScrollSearch and PitSearch. It could be cleaner if two implementations were separate, e.g., no need for comments that some piece is needed for one, but not the other. The downside is that some code needs to be duplicated, but important bits could be moved to a shared function.

Reply:
b3de51cb00d5702633c58e69e64e7639306a2c97 571d299224628934b32f9af55bf42a69b6028095

Reply:
Thanks! will test! Do you plan to add functional tests for the config option or do you think we should be good with unit tests here?

Reply:
I would like to add one more unit test for `generate_csv.ts`. I hesitate to create a functional test, since it would mean adding a new config for the functional test server to launch with the correct kibana.yml setting to test this feature. I'm still on the fence about it and can be convinced it might be needed.